### PR TITLE
[TRAFODION-3280] Reduce path length in Trafodion for improved performance and scalability

### PR DESCRIPTION
**Pull request by selvaganesang**
This PR fixes the following issues to improve performance.
1. When the session is dropped for the default context, we were unnecessarily dropping HBase connections.
2. There were 2 HBase connections per mxosrvr - one for the default context and another for repository context. Now it has been changed to one HBase connection per process.
3. cli_globals::currContext() is now optimized for default context.
4. Changes to ensure that logging related objects are created and the corresponding functions are called only when the logging Level is set higher than the recording level,

### CODE REVIEW

File: core/sql/cli/Globals.h [hunk 1]
```diff
@@ -89,6 +89,7 @@ class ExProcessStats;
 class CliGlobals;
 class CLISemaphore;
 class HBaseClient_JNI;
+class MonarchClient_JNI;
```
Comment (DaveBirdsall) [inline core/sql/cli/Globals.h]: Was this line intended?

File: core/sql/cli/Globals.h [hunk 2]
```diff
@@ -510,6 +523,9 @@ inline
   LmLanguageManagerC *langManC_;
   LmLanguageManagerJava *langManJava_;
   NABoolean espProcess_;
+  HBaseClient_JNI *hbaseClientJNI_;
+  HBaseClient_JNI *bigtableClientJNI_;
+  MonarchClient_JNI *monarchClientJNI_;
```
Comment (DaveBirdsall) [inline core/sql/cli/Globals.h]: Not clear to me what bigtableClientJNI_ and monarchClientJNI_ are intended for.

Comment (selvaganesang) [reply core/sql/cli/Globals.h]: I will remove it in my next PR unless there is a need for another commit

File: core/sql/cli/Globals.h (hunk 1 above)
Comment (selvaganesang) [reply core/sql/cli/Globals.h]: No. I will remove it in my next PR.

**Comment by selvaganesang** [comment]
@DaveBirdsall I have pushed in a commit to take care of your comments. Can you please resolve the conversations and approve this PR

**Pull request merged**